arroyo-types/src/functions/hash.rs

```diff
@@ -0,0 +1,45 @@
+use base64::{engine::general_purpose::STANDARD_NO_PAD as encoder, Engine};
+use md5::{Digest, Md5};
+
+// would like to make this parameterized based on the hasher, since everything else looks the same
+// but I couldn't quite figure out what the common trait was between all of them...
+// I'll ask later if anyone has ideas.
+pub fn md5(argument: String) -> String {
+    let mut hasher = Md5::new();
+    hasher.update(argument.as_bytes());
+    let res = hasher.finalize();
+    // chose to encode as base64 to encode bytes to string, but not sure if that's correct
```

Comment:
From the [postgres docs](https://www.postgresql.org/docs/11/functions-binarystring.html), md5 is a hex-encoded text representation, while the shaXXX functions return a byte array.

> Note that for historic reasons, the function md5 returns a hex-encoded value of type text whereas the SHA-2 functions return type bytea. Use the functions encode and decode to convert between the two, for example encode(sha256('abc'), 'hex') to get a hex-encoded text representation.

DataFusion seems to follow this convention, so it would probably be good if we did as well.